src/schema-builder/RdbmsSchemaBuilder.ts

```diff
@@ -99,17 +96,24 @@ export class RdbmsSchemaBuilder implements SchemaBuilder {
         }
     }
 
+    /**
+     * If the schema contains views, create the typeorm_metadata table if it doesn't exist yet
+     */
+    async createMetadataTableIfNecessary(): Promise<void> {
+        if (this.viewEntityToSyncMetadatas.length > 0) {
+            this.queryRunner = this.queryRunner || this.connection.createQueryRunner();
```

Comment:
`|| this.connection.createQueryRunner()` this code can be very dangerous since its not clear here where it gets **released**. Use `this.queryRunner` if it's available, or pass it to `createMetadataTableIfNecessary` method from the places where you call this method.

Reply:
On a side note, with both the current implementation [1]/[2] as well as this PR the migration table is created when **generating** migrations. Is this intended behavior?

[1] https://github.com/typeorm/typeorm/blob/a92120ac8821461055993a36ddba17d3466de925/src/commands/MigrationGenerateCommand.ts#L105
[2] https://github.com/typeorm/typeorm/blob/a92120ac8821461055993a36ddba17d3466de925/src/schema-builder/RdbmsSchemaBuilder.ts#L116-L125

Reply:
> the migration table is created when generating migrations. Is this intended behavior?

Probably it shouldn't be inside `log` method, since you are creating metadata table in `executePendingMigrations`, right?